### PR TITLE
[READY] Superior Water Melon

### PR DESCRIPTION
**Pull request by ghost**
## About The Pull Request

This PR adds a new melon type, the _Superior_ Watermelon! This watermelon exhibits healing properties as it contains Protozine, Omnizine's more retarded brother! The plant is harder to grow and more unstable, but the healing that its juice carries makes the endeavour worthwhile for any Botanist who tries.

Got this idea from this man:
image

## Why It's Good For The Game

More variety, as usual, is always a better thing. Plus holy melons are a rare mutation and usually are only spawned by the gods. This will join the roster of healing plants that contain healing properties and reagents. Deus is still better, remember that Omnizine is the non-retarded brother.
image

## Changelog
:cl:
add: Added the melon
/:cl:

### CODE REVIEW

File: code/modules/hydroponics/grown/melon.dm [hunk 1]
```diff
@@ -76,3 +76,34 @@
 	to_chat(user, "<span class='warning'>[src] rapidly turns into ash!</span>")
 	qdel(src)
 	new /obj/effect/decal/cleanable/ash(drop_location())
+
+//Superior Melon
+/obj/item/seeds/superwatermelon
```
Comment (Fikou) [inline code/modules/hydroponics/grown/melon.dm:81]: this should be a child of watermelon seeds

File: code/modules/hydroponics/grown/melon.dm [hunk 2]
```diff
@@ -76,3 +76,34 @@
...
+	lifespan = 20
+	endurance = 20
+	instability = 80
+	growing_icon = 'icons/obj/hydroponics/growing_fruits.dmi'
+	icon_dead = "superwatermelon-dead"
+	genes = list(/datum/plant_gene/trait/repeated_harvest)
+	mutatelist = list(/obj/item/seeds/watermelon/holy)
+	reagents_add = list(/datum/reagent/water = 0.2, /datum/reagent/consumable/nutriment/vitamin = 0.04, /datum/reagent/consumable/nutriment = 0.2, /datum/reagent/medicine/omnizine/protozine = 0.5)
+
+/obj/item/reagent_containers/food/snacks/grown/superwatermelon
```
Comment (Fikou) [inline code/modules/hydroponics/grown/melon.dm:98]: this too

File: code/modules/hydroponics/grown/melon.dm [hunk 3]
```diff
@@ -76,3 +76,34 @@
...
+	plantname = "Superior Watermelon Vines"
+	product = /obj/item/reagent_containers/food/snacks/grown/superwatermelon
+	rarity = 10
+	lifespan = 20
+	endurance = 20
+	instability = 80
+	growing_icon = 'icons/obj/hydroponics/growing_fruits.dmi'
+	icon_dead = "superwatermelon-dead"
+	genes = list(/datum/plant_gene/trait/repeated_harvest)
+	mutatelist = list(/obj/item/seeds/watermelon/holy)
```
Comment (Fikou) [inline code/modules/hydroponics/grown/melon.dm:95]: is this intended

Comment (ghost) [reply code/modules/hydroponics/grown/melon.dm:95]: It is, the progression should be melon -> Super -> Holy

Comment (Fikou) [reply code/modules/hydroponics/grown/melon.dm:95]: but its melon -> holy or super

Comment (ghost) [reply code/modules/hydroponics/grown/melon.dm:95]: ill edit the melon to mutate ONLY to super

**Comment by Ratvarr** [comment]
i am the melon man

File: code/modules/hydroponics/grown/melon.dm [hunk 4]
```diff
@@ -12,7 +12,7 @@
 	growing_icon = 'icons/obj/hydroponics/growing_fruits.dmi'
 	icon_dead = "watermelon-dead"
 	genes = list(/datum/plant_gene/trait/repeated_harvest)
-	mutatelist = list(/obj/item/seeds/watermelon/holy)
+	mutatelist = list(/obj/item/seeds/watermelon/holy, /obj/item/seeds/superwatermelon)
```
Comment (ATH1909) [inline code/modules/hydroponics/grown/melon.dm:15]: you forgot to do what you said that you'd do, and watermelons are still able to mutate directly into holymelons

**Comment by craigcalef** [comment]
Why melon -> super -> holy instead of melon -> super or holy?

**Pull request closed without merging**